ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateTransition.java

```diff
@@ -117,10 +117,10 @@ public BeaconStateWithCache initiate(BeaconStateWithCache state, BeaconBlock blo
    * @throws BlockProcessingException
    */
   private void process_block(
-      BeaconStateWithCache state, BeaconBlock block, boolean validate_state_root)
+      BeaconStateWithCache state, BeaconBlock block, boolean validateStateRootAndSignatures)
       throws BlockProcessingException {
-    process_block_header(state, block, validate_state_root);
-    process_randao(state, block.getBody());
+    process_block_header(state, block, validateStateRootAndSignatures);
+    process_randao(state, block.getBody(), validateStateRootAndSignatures);
```

Comment:
Added a validate boolean here, so we can skip randao signature verification in tests where we generate a block using the wrong proposer.

Reply:
nice